service/r2/store/store.go

```diff
@@ -38,8 +39,11 @@ type Store interface {
 	// DeleteNamespace deletes the namespace for the given namespace ID.
 	DeleteNamespace(namespaceID string, uOpts UpdateOptions) error
 
-	// FetchRuleSet fetches the ruleset for the given namespace ID.
-	FetchRuleSet(namespaceID string) (*models.RuleSetSnapshotView, error)
+	// FetchRuleSet fetch ruleset from KV.
+	FetchRuleSet(namespaceID string) (rules.RuleSet, error)
```

Comment:
I think if you move the rule mutation logic into `m3metrics`, you don't need this API.

Reply:
I think what you mean here is that rather than fetching, calling the apply method and writing there could be one method?

That could work, but all of the methods in this class follow the pattern of fetch -> mutate -> write. Perhaps this is worth changing the paradigm here but I am not sure it needs to happen before this diff.

Reply:
Yeah what I meant is that the convention is to simply call `s.ruleStore.ReadRuleSet(namespaceID)` in the store implementation. It doesn't seem like we need to call this outside the store package, so I'd prefer to remove it from the interface to keep the API surface area small.

Reply:
Good call. Does not need to be public right now.